### ISSUE TITLE
array_merge on array to translate string only translating first one

### ISSUE THREAD

**Issue by aldorr** [polylang/polylang#314]
## Prerequisites

- [x] This is not a usage question (Those should be directed to the [community supported forum](https://wordpress.org/support/plugin/polylang), unless this is a question about Polylang Pro in which case you should [use the helpdesk](https://polylang.pro/support/)).
- [x] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [x] The issue still exists against the latest `master` branch of Polylang and the latest WordPress version.
- [x] This bug happens with only Polylang plugin active
- [x] This bug happens with a default WordPress theme active
- [x] I can reproduce this bug consistently

## Steps to reproduce the issue

1. Register Strings
2. Translate in Dashboard
3. Feed array via `map_array` to `pll__`
4. New array has only first string translated

## Expected behavior and actual behavior

When I use the following code:
`
            $translated_array = array_map("pll__", $words_array);
`
where `$words_array`
is something like:
`
Array
(
    [0] => Körpertherapie
    [1] =>  Kunsttherapie
    [2] =>  Bewegungs-Therapie
)
`
It seems, the whole array should be translated. That is, each value in the array.

But I get a new array, with the first value translated, the rest not.
`
Array
(
    [0] => Thérapie corporelle
    [1] =>  Kunsttherapie
    [2] =>  Bewegungs-Therapie
)
`

The string "Kunsttherapie" is registered and translated. "Bewegungs-Therapie" as well.
When these strings are the first ones in the array, they also translate. The rest, not.

Any ideas, how this happens?

Thanks in advance.

**Comment by chesio** [comment]
Hi, I cannot confirm. I just tested the same with Polylang 2.5.0 and all strings in array were translated.

How exactly do you register you strings?

**Comment by Chouby** [comment]
Closing this issue as there was no answer.

**Comment by aldorr** [comment]
Thanks, and sorry for getting back _so_ late. I figured it out. It was my mistake, because sometimes I was getting an array and sometimes a string, depending on how many values.
I fixed it with an array check, like this:
`if ( is_array( $words ) ) {
	$words_translated = array_map("pll__", $words);
	$words_translated = implode(  ', ', $words_translated );
} else {
	$words_translated = pll__($words);
}
return $words_translated;`
That way I always get a comma separated list or a single word.
